applications/RANSApplication/tests/cpp/test_incompressible_potential_flow_conditions.cpp

```diff
@@ -0,0 +1,282 @@
+
+    // Test:
+    Matrix LHS, ref_LHS(2, 2, 0.0);
+    auto& r_condition = r_model_part.Conditions().front();
+
+    r_condition.SetValue(RANS_IS_INLET, 0);
+    r_condition.Initialize();
+    r_condition.CalculateLeftHandSide(
+        LHS, static_cast<const ProcessInfo>(r_model_part.GetProcessInfo()));
+
+    KRATOS_CHECK_MATRIX_NEAR(LHS, ref_LHS, 1e-12);
+
+    r_condition.SetValue(RANS_IS_INLET, 1);
+    r_condition.Initialize();
+    r_condition.CalculateLeftHandSide(
+        LHS, static_cast<const ProcessInfo>(r_model_part.GetProcessInfo()));
+
+    KRATOS_CHECK_MATRIX_NEAR(LHS, ref_LHS, 1e-12);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(RansIncompressiblePotentialFlowVelocityInlet2D2N_CalculateRightHandSide,
+                          KratosRansFastSuite)
+{
+    // Setup:
+    Model model;
+    auto& r_model_part = RansIncompressiblePotentialFlowVelocityInlet2D2N_SetUp(model);
+
+    // Test:
+    Vector RHS, ref_RHS(2, 0.0);
+    auto& r_condition = r_model_part.Conditions().front();
+
+    r_condition.SetValue(RANS_IS_INLET, 0);
+    r_condition.Initialize();
+    r_condition.CalculateRightHandSide(
+        RHS, static_cast<const ProcessInfo>(r_model_part.GetProcessInfo()));
+
+    r_condition.SetValue(RANS_IS_INLET, 1);
+    r_condition.Initialize();
+    r_condition.CalculateRightHandSide(
+        RHS, static_cast<const ProcessInfo>(r_model_part.GetProcessInfo()));
+    ref_RHS[0] = -7.4912949377611748e-01;
+    ref_RHS[1] = -7.4912949377611748e-01;
+
+    KRATOS_CHECK_VECTOR_NEAR(RHS, ref_RHS, 1e-12);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(RansIncompressiblePotentialFlowPressureBodyForce2D2N_EquationIdVector,
+                          KratosRansFastSuite)
+{
+    // Setup:
+    Model model;
+    auto& r_model_part = RansIncompressiblePotentialFlowPressureBodyForce2D2N_SetUp(model);
+
+    // Test:
+    RansApplicationTestUtilities::TestEquationIdVector<ModelPart::ConditionsContainerType>(r_model_part);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(RansIncompressiblePotentialFlowPressureBodyForce2D2N_GetDofList,
+                          KratosRansFastSuite)
+{
+    // Setup:
+    Model model;
+    auto& r_model_part = RansIncompressiblePotentialFlowPressureBodyForce2D2N_SetUp(model);
+
+    // Test:
+    RansApplicationTestUtilities::TestGetDofList<ModelPart::ConditionsContainerType>(
+        r_model_part, PRESSURE_POTENTIAL);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(RansIncompressiblePotentialFlowPressureBodyForce2D2N_CalculateLocalSystem,
+                          KratosRansFastSuite)
+{
+    // Setup:
+    Model model;
+    auto& r_model_part = RansIncompressiblePotentialFlowPressureBodyForce2D2N_SetUp(model);
+
+    // Test:
+    Matrix LHS, ref_LHS(2, 2, 0.0);
+    Vector RHS, ref_RHS(2);
+    auto& r_condition = r_model_part.Conditions().front();
+    r_condition.Initialize();
+    r_condition.CalculateLocalSystem(
+        LHS, RHS, static_cast<const ProcessInfo>(r_model_part.GetProcessInfo()));
```

Comment:
```suggestion
   const auto& r_process_info = r_model_part.GetProcessInfo();
    r_condition.CalculateLocalSystem(
        LHS, RHS, r_process_info);
```
Avoid cast's

Reply:
I need casts, because otherwise It will call the empty CalculateLocalSystem of base class `Condition`. At least I need them casts untill @philbucher removes non const versions from the base class. Do you know a workaround?

Reply:
I think that getting the `ProcessInfo` with the `const` specifier as in the suggestion should work too...

Reply:
done :)